go/tools/builders/nogo_main.go

```diff
@@ -229,40 +226,59 @@ func checkAnalysisResults(c chan result, fset *token.FileSet) error {
 			if len(config.applyTo) > 0 {
 				// This analysis applies exclusively to a set of files.
 				include = false
-				for pattern := range config.applyTo {
-					if matched, err := regexp.MatchString(pattern, filename); err == nil && matched {
+				for _, pattern := range config.applyTo {
+					if pattern.MatchString(filename) {
 						include = true
+						break
 					}
 				}
 			}
-			for pattern := range config.whitelist {
-				if matched, err := regexp.MatchString(pattern, filename); err == nil && matched {
-					include = false
+			if include {
+				for _, pattern := range config.whitelist {
+					if pattern.MatchString(filename) {
+						include = false
+						break
+					}
 				}
 			}
 			if include {
-				analysisFindings = append(analysisFindings, finding)
+				findings = append(findings, finding)
 			}
 		}
 	}
-	if len(analysisFindings) == 0 {
+	if len(findings) == 0 && len(errs) == 0 {
 		return nil
 	}
-	sort.Slice(analysisFindings, func(i, j int) bool {
-		return analysisFindings[i].Pos < analysisFindings[j].Pos
+
+	sort.Slice(findings, func(i, j int) bool {
+		return findings[i].Pos < findings[j].Pos
 	})
-	var errMsg bytes.Buffer
-	for i, f := range analysisFindings {
-		if i > 0 {
-			errMsg.WriteByte('\n')
-		}
-		errMsg.WriteString(fmt.Sprintf("%s: %s\n", fset.Position(f.Pos), f.Message))
+	errMsg := &bytes.Buffer{}
+	sep := ""
+	for _, err := range errs {
```

Comment:
I'm afraid I disagree.

I think of these errors like internal compiler errors: they should be quite rare, and I would much rather have a build fail in this case.

If we only print a failure, I don't think anyone will see it. Bazel is very chatty, and warning messages can easily get buried in the output. Also, if an error comes up in CI, it will be hidden in a log for something that's apparently working. Personally, I almost never read logs of passing builds and tests.

Failing the build is the best way to bring the failure to the developer's attention. It should be possible to work around failures by editing nogo.json. We may want to add that to the error message so it's clear how to proceed.

I feel strongly about this because nogo will be used for security checks, and it's important to fail closed. This might be far fetched, but I could imagine a malicious author submitting a PR that would fail a security check in CI, but they've found a clever way to trigger an error in an analysis. We should fail the build in that case.